source/common/ssl/context_impl.h

```diff
@@ -37,8 +36,8 @@ struct SslStats {
   ALL_SSL_STATS(GENERATE_COUNTER_STRUCT, GENERATE_GAUGE_STRUCT, GENERATE_TIMER_STRUCT)
 };
 
-typedef CSmartPtr<SSL, SSL_free> SslConPtr;
-typedef CSmartPtr<X509, X509_free> X509Ptr;
+typedef bssl::UniquePtr<SSL> SslConPtr;
+typedef bssl::UniquePtr<X509> X509Ptr;
```

Comment:
Since pointers are so common we are using the type aliasing to avoid excess typing, so I think this is fine. FWIW, we do typedef std::vector in a few places to avoid a ton of typing, but there is no hard rule.